crates/sui-indexer/src/indexer_reader.rs

```diff
@@ -555,20 +555,45 @@ impl IndexerReader {
     pub async fn get_owned_objects_in_blocking_task(
         &self,
         address: SuiAddress,
-        object_type: Option<String>,
+        filter: Option<SuiObjectDataFilter>,
         cursor: Option<ObjectID>,
         limit: usize,
     ) -> Result<Vec<StoredObject>, IndexerError> {
+        let object_types = Self::extract_struct_filters(filter)?;
         self.spawn_blocking(move |this| {
-            this.get_owned_objects_impl(address, object_type, cursor, limit)
+            this.get_owned_objects_impl(address, object_types, cursor, limit)
         })
         .await
     }
 
+    fn extract_struct_filters(
+        filter: Option<SuiObjectDataFilter>,
+    ) -> Result<Option<Vec<String>>, IndexerError> {
+        if filter.is_none() {
+            return Ok(None);
+        }
+        match filter.unwrap() {
+            SuiObjectDataFilter::StructType (struct_tag ) => Ok(Some(vec![struct_tag.to_string()])),
+            SuiObjectDataFilter::MatchAny(filters) => {
+                filters.iter().map(|filter| {
+                    match filter {
+                        SuiObjectDataFilter::StructType (struct_tag ) => Ok(struct_tag.to_string()),
```

Comment:
Can we double check that this string form is the same as is stored in the DB? I know we have some issues with canonical type format in the db

Reply:
Yes I was actually using to_canonical_string initially but it didn't work while testing and changing it to to_string worked. But good point, I will double check the write side.

Reply:
oh the PR is still in flux: https://github.com/MystenLabs/sui/pull/14382

Reply:
Will merge this one and follow up once that PR merges.